### PR TITLE
NullRef in Darc's DisambiguateAssetsAsync

### PR DESCRIPTION
**Pull request by epananth**
Issue -> https://github.com/dotnet/arcade/issues/6651

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/src/DarcLib/IRemote.cs [hunk 1]
```diff
@@ -265,6 +267,14 @@ public interface IRemote
 
         #region Repo/Dependency Operations
 
+        /// <summary>
+        /// Get the tools.dotnet section of the global.json from a target repo URI
```
Comment (mmitche) [inline src/Microsoft.DotNet.Darc/src/DarcLib/IRemote.cs:271]: Can you fix up the comments here (I was just copying/pasting when unblocking the source build folks)

**Comment by sunandabalu** [comment]
@mmitche Could pls review the PR?

**Review by mmitche — APPROVED** [review]

**Comment by mmitche** [comment]
LGTM thanks!

**Comment by riarenas** [comment]
It looks like this PR is causing test failures: https://github.com/dotnet/core-eng/issues/11693 Looks like it made one of the coherency tests flaky?

**Pull request merged**